C/Cpp_include/c4Observer.hh

```diff
@@ -42,6 +44,7 @@ struct C4CollectionObserver : public fleece::InstanceCounted, C4Base {
 
     /// Metadata of a change recorded by C4CollectionObserver. (Equivalent to C4CollectionChange.)
     struct Change {
+        CollectionSpec collection;      ///< The collection the document belongs to
```

Comment:
In this case, the client code could supply the "context" in 
c4docobs_create(C4Database *db,
                                    C4Slice docID,
                                    C4DocumentObserverCallback callback,
                                    void *context) noexcept
which will be passed back to the client in callback.

Reply:
Usually the context is already being used for storing some kind of other reference (Mapping to a Java or .NET object is common).  This does not translate all the way up to the user.

Reply:
Users probably may store the collection info in the context. Each collection commands a unique context object. This can save a lot of complexity in the LiteCore.